### ISSUE TITLE
[Loki] Queries from pachctl with long time ranges 500 Loki

### ISSUE THREAD

**Issue by echohack** [pachyderm/pachyderm#5540]
**What happened?**:

When using `pachctl logs` and querying Loki, long time ranges cause Loki to use all of the available RAM on the box and 500s Loki. There appears to be long query times included in the `start` and `end` fields that may be the cause of this issue.

**What you expected to happen?**:

Queries to Loki from `pachctl logs` should return normally.

Loki's API expects nanosecond unix epoch time in the `start` and `end` fields. By default, the default `start` time is 1 hour ago and `end` time is `now()`. I would expect that the query sent from pachyderm is similarly constrained to 1 hour ago by default, with a configurable start and end time, plus some kind of check to ensure that this amount of time is not crazy (like 1 year absolute max or something? Needs research on a good "ceiling" value).

**How to reproduce it (as minimally and precisely as possible)?**:

Do a simple query from `pachctl logs` with very long time ranges.

**Logs from Loki**:

```
level=error ts=2020-12-14T20:14:56.030291933Z caller=chunk_store.go:533 org_id=fake traceID=3efba9d0551208d1 msg="error querying storage" err=timeout
level=info ts=2020-12-14T20:14:56.030502108Z caller=metrics.go:81 org_id=fake traceID=3efba9d0551208d1 latency=slow query="{pipelineName=\"tempSoil_level1_group\", container=\"user\"} |= \"ee632ed036424591a1be651f724e3c96\"" query_type=filter range_type=range length=2334261h25m34.812397546s step=9337h2m44s duration=5m40.087670632s status=500 throughput_mb=0 total_bytes_mb=0
level=warn ts=2020-12-14T20:14:56.030595594Z caller=logging.go:71 traceID=3efba9d0551208d1 msg="GET /loki/api/v1/query_range?direction=FORWARD&end=1607976555941052394&limit=5000&query=%7BpipelineName%3D%22tempSoil_level1_group%22%2C+container%3D%22user%22%7D+%7C%3D+%22ee632ed036424591a1be651f724e3c96%22&start=-6795364578871345152 (500) 5m40.087982343s Response: \"timeout\\n\" ws: false; User-Agent: Go-http-client/1.1; "
level=error ts=2020-12-14T20:15:00.990604422Z caller=chunk_store.go:533 org_id=fake traceID=6cdcb848d9d873d8 msg="error querying storage" err=timeout
level=info ts=2020-12-14T20:15:00.990812924Z caller=metrics.go:81 org_id=fake traceID=6cdcb848d9d873d8 latency=slow query="{pipelineName=\"tempSoil_level1_group\", container=\"user\"} |= \"ee632ed036424591a1be651f724e3c96\"" query_type=filter range_type=range length=2334261h25m40.445507774s step=9337h2m44s duration=5m39.414677798s status=500 throughput_mb=0 total_bytes_mb=0
level=warn ts=2020-12-14T20:15:00.990908156Z caller=logging.go:71 traceID=6cdcb848d9d873d8 msg="GET /loki/api/v1/query_range?direction=FORWARD&end=1607976561574162622&limit=5000&query=%7BpipelineName%3D%22tempSoil_level1_group%22%2C+container%3D%22user%22%7D+%7C%3D+%22ee632ed036424591a1be651f724e3c96%22&start=-6795364578871345152 (500) 5m39.414988248s Response: \"timeout\\n\" ws: false; User-Agent: Go-http-client/1.1; "
```
Note the start and end times here are both very large (and also negative).

```
end=1607976555941052394
start=-6795364578871345152
```

I suspect we need to constrain the `start` and `end` times here, but also possibly other places:

https://github.com/pachyderm/pachyderm/blob/bb1e25f60f11418253b60dbd22b4e0d36139fb2c/src/server/pps/server/api_server.go#L1759

**Anything else we need to know?**:

**Environment?**:

```
pachctl: 1.11.8
pachd: 1.11.8
grafana/loki:2.0.0
grafana/promtail:2.0.0
```

Refs:

Loki HTTP API: https://grafana.com/docs/loki/latest/api/#get-lokiapiv1query_range

LogQL with Loki 2.0: https://grafana.com/blog/2020/10/28/loki-2.0-released-transform-logs-as-youre-querying-them-and-set-up-alerts-within-loki/

**Comment by brokenjacobs** [comment]
Thanks for logging this in for me @echohack

**Comment by brokenjacobs** [comment]
I don't know what version of loki you want to support, but 2.0 has some nice json parsing features in LogQL now, so you could search the specific field for the job id.

Another option is to have promtail parse the logs and add a jobid field, but cardinality would be high.

It's worth noting that the query being passed right now works 'just fine' if you shorten the time scale.

**Comment by nitinjainsj** [comment]
This issue is fixed #5629
